polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1140,8 +1140,9 @@ MLIRScanner::emitSYCLOps(const clang::Expr *Expr,
           !RD->getName().empty()) {
         std::string Name =
             MLIRScanner::getMangledFuncName(*Func, Glob.getCGM());
-        return Builder.create<mlir::sycl::SYCLConstructorOp>(Loc, RD->getName(),
-                                                             Name, Args);
+        SmallVector<Value> RemainderArgs(Args.begin() + 1, Args.end());
+        return Builder.create<mlir::sycl::SYCLConstructorOp>(
+            Loc, Args[0], RemainderArgs, RD->getName(), Name);
```

Comment:
Can we use an `ArrayRef` instead of a `SmallVector` to avoid memory allocation?

Reply:
ok